app/scripts/migrations/148.ts

```diff
@@ -0,0 +1,87 @@
+import { toHex } from '@metamask/controller-utils';
+import { getErrorMessage, hasProperty, isObject } from '@metamask/utils';
+import { cloneDeep } from 'lodash';
+
+export const version = 148;
+
+/**
+ * This migration resets sessionData and isSignedIn if using the old sessionData state shape.
+ *
+ * @param originalVersionedData - Versioned MetaMask extension state, exactly what we persist to dist.
+ * @param originalVersionedData.meta - State metadata.
+ * @param originalVersionedData.meta.version - The current state version.
+ * @param originalVersionedData.data - The persisted MetaMask state, keyed by controller.
+ * @returns Updated versioned MetaMask extension state.
+ */
+export async function migrate(originalVersionedData: {
+  meta: { version: number };
+  data: Record<string, unknown>;
+}) {
+  const versionedData = cloneDeep(originalVersionedData);
+  versionedData.meta.version = version;
+  try {
+    versionedData.data = transformState(versionedData.data);
+  } catch (error) {
+    global.sentry?.captureException?.(
+      new Error(`Migration #${version}: ${getErrorMessage(error)}`),
+    );
+  }
+  return versionedData;
+}
+
+function transformState(state: Record<string, unknown>) {
+  const newState = cloneDeep(state);
+
+  if (!hasProperty(newState, 'NetworkController')) {
+    throw new Error(`newState.NetworkController must be present`);
+  }
+
+  if (!isObject(newState.NetworkController)) {
+    throw new Error(
+      `state.NetworkController must be an object, but is: ${typeof newState.NetworkController}`,
+    );
+  }
+  if (
+    !hasProperty(newState.NetworkController, 'networkConfigurationsByChainId')
+  ) {
+    throw new Error(
+      `state.NetworkController.networkConfigurationsByChainId must be present`,
+    );
+  }
+
+  if (!isObject(newState.NetworkController.networkConfigurationsByChainId)) {
+    throw new Error(
+      `state.NetworkController.networkConfigurationsByChainId must be an object, but is: ${typeof newState
+        .NetworkController.networkConfigurationsByChainId}`,
+    );
+  }
+
+  const { networkConfigurationsByChainId } = newState.NetworkController;
+
+  for (const [chainId, networkConfiguration] of Object.entries(
+    networkConfigurationsByChainId,
+  )) {
+    const chainIdAsHex = toHex(chainId);
```

Comment:
do we need to guard against chainId not being able to be converted to hex for some reason?

Reply:
Unsure what the best course of action here is. Aborting leaves us in a corrupted state, but we can't just naively delete the networkConfiguration in case there are other places in state that reference a networkClientId that's part of that networkConfiguration and doing that seems out of scope for this particular migration.

I guess if we just let `toHex` throw and abort that's fine. If we get a non integer value in the chainId field, then we have bigger problems.